dolphin/task_manager/tasks/task.py

```diff
@@ -42,11 +41,13 @@ def _set_synced_after(func, *args, **kwargs):
                 LOG.warn('Storage %s not found when set synced'
                          % self.storage_id)
             else:
-                storage[constants.DB.DEVICE_SYNC_STATUS] = utils.set_bit(
-                    storage[constants.DB.DEVICE_SYNC_STATUS],
-                    resource_type,
-                    constants.SyncStatus.SYNCED)
-                db.storage_update(self.context, self.storage_id, storage)
+                # One sync task done, sync status minus 1
+                # When sync status get to 0
+                # means all the sync tasks are completed
+                if storage[constants.DB.DEVICE_SYNC_STATUS] > 0:
+                    storage[constants.DB.DEVICE_SYNC_STATUS] -= 1
```

Comment:
Suggest to rename constants.DB.DEVICE_SYNC_STATUS as constants.DB.DEVICE_SYNCING_RESOURCE_COUNT for better readability.. beacuse it is a counter represets how many resources are in syncing state.

Reply:
suggest:
```
if storage['sync_status'] != constants.SyncStatus:
    storage['sync_status'] -= 1
```

Reply:
Done